test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/testproviders/PersistenceTestProvider.java

```diff
@@ -85,6 +90,134 @@ public void testControllerPersistence(WorkbenchKieServerScenario deploymentScena
+            // Check if old user can connect
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertAlso(isUserAuthorizedInWorkbench(workbenchUrl, username, oldPassword));
+                softly.assertAlso(isUserAuthorizedInKieServer(kieServerUrl, username, oldPassword));
+            });
+
+            logger.info("Change username to {} and password to {} in credential secret", username, NEW_PASSWORD);
+            deploymentScenario.changeUsernameAndPassword(username, NEW_PASSWORD);
+            //workbenchDeployment.changeUsernameAndPassword(NEW_USERNAME, NEW_PASSWORD);
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertAlso(isUserUnauthorizedInWorkbench(workbenchUrl, username, oldPassword));
+
+                softly.assertAlso(isUserAuthorizedInWorkbench(workbenchUrl, username, NEW_PASSWORD));
+
+                softly.assertAlso(isUserUnauthorizedInKieServer(kieServerUrl, username, oldPassword));
+
+                softly.assertAlso(isUserAuthorizedInKieServer(kieServerUrl, username, NEW_PASSWORD));
+            });
+        } finally {
+            deploymentScenario.changeUsernameAndPassword(username, oldPassword);
+            kieControllerClient.deleteContainerSpec(serverInfo.getServerId(), containerId);
+        }
+    }
+
+    // Verifies https://issues.redhat.com/browse/RHPAM-2777
+    public void testAdminUserNameAndPasswordChange(WorkbenchKieServerScenario deploymentScenario) {
+        final String NEW_USERNAME = "newadminusername";
+        final String NEW_PASSWORD = "newpassword4321!";
+
+        final String oldUsername = deploymentScenario.getWorkbenchDeployment().getUsername();
+        final String oldPassword = deploymentScenario.getWorkbenchDeployment().getPassword();
+        String containerId = "testAdminUserNameAndPasswordChange";
+        final String workbenchUrl = deploymentScenario.getWorkbenchDeployment().getUrl().toString();
+        final String kieServerUrl = deploymentScenario.getKieServerDeployment().getUrl().toString();
+
+        KieServerControllerClient kieControllerClient = KieServerControllerClientProvider.getKieServerControllerClient(deploymentScenario.getWorkbenchDeployment());
+        KieServicesClient kieServerClient = KieServerClientProvider.getKieServerClient(deploymentScenario.getKieServerDeployment());
+
+        KieServerInfo serverInfo = kieServerClient.getServerInfo().getResult();
+        try {
+            WorkbenchUtils.saveContainerSpec(kieControllerClient, serverInfo.getServerId(), serverInfo.getName(), containerId, containerId, Kjar.DEFINITION, KieContainerStatus.STARTED);
+            // Check if old user can connect
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertAlso(isUserAuthorizedInWorkbench(workbenchUrl, oldUsername, oldPassword));
+                softly.assertAlso(isUserAuthorizedInKieServer(kieServerUrl, oldUsername, oldPassword));
+            });
```

Comment:
I don't think we need to assert old users can connect as if not, it would have failed before (to kie server when connecting to business central or when deploying the container).